src/driver/ixgbe.c

```diff
@@ -432,6 +675,26 @@ uint32_t ixgbe_rx_batch(struct ixy_device* ixy, uint16_t queue_id, struct pkt_bu
 		set_reg32(dev->addr, IXGBE_RDT(queue_id), last_rx_index);
 		queue->rx_index = rx_index;
 	}
+
+	if (interrupts_enabled) {
+		bool int_en = interrupt->interrupt_enabled;
+		interrupt->rx_pkts += buf_index;
+
+		uint64_t diff = monotonic_time() - interrupt->last_time_checked;
```

Comment:
i don't like running monotonic_time for every poll; sure, it should be a simple vdso call that should just be rdtsc and it should be fast.

but can we quantify the overhead here? i.e. run a performance evaluation at a lower clock speed and compare interrupts enabled but dynamically disabled due to high load vs. interrupts completely disabled at initilization?

Reply:
ok I will try to find a better solution for that then using `monotonic_time` :)

Reply:
not saying monotonic_time is necessarily wrong, a benchmark would be interesting.

for your thesis: how does the kernel ixgbe driver handle this? is it only based on last call there?

would it be possible to only check the time every X calls?

Reply:
I changed it now to only check the monotonic_time every X calls and now there is no visible performance degradation as if interrupts are disabled :) Before there was a 2-5% increase in CPU utilization compared to now.
For my thesis I will check how the kernel driver is doing this :)